sound/soc/sof/ipc.c

```diff
@@ -433,13 +433,22 @@ static void ipc_trace_message(struct snd_sof_dev *sdev, u32 msg_id)
 
 static void ipc_period_elapsed(struct snd_sof_dev *sdev, u32 msg_id)
 {
-	struct snd_soc_component *scomp = sdev->sof_audio_data->component;
 	struct snd_sof_pcm_stream *stream;
 	struct sof_ipc_stream_posn posn;
+	struct snd_sof_client *client;
 	struct snd_sof_pcm *spcm;
 	int direction;
 
-	spcm = snd_sof_find_spcm_comp(scomp, msg_id, &direction);
+	/* search all clients */
+	list_for_each_entry(client, &sdev->client_list, list) {
+		if (client->type != SOF_CLIENT_AUDIO)
+			continue;
+
+		spcm = snd_sof_find_spcm_comp(&client->pdev->dev, msg_id,
+					      &direction);
+		if (spcm)
+			break;
+	}
```

Comment:
same comment as above for dais, can we skip the search and somehow keep track of the spcm on startup?